pages/profile.tsx

```diff
@@ -0,0 +1,191 @@
+  const [csrfToken, setCsrfToken] = useState<string>("")
+  const [traits, setTraits] = useState<any>()
+  const [profile, setProfile] = useState<any>({})
+
+  useEffect(() => {
+    // If the router is not ready yet, or we already have a flow, do nothing.
+    if (!router.isReady || flow) {
+      return
+    }
+
+    // If ?flow=.. was in the URL, we fetch it
+    if (flowId) {
+      ory
+        .getSettingsFlow({ id: String(flowId) })
+        .then(({ data }) => {
+          setFlow(data)
+        })
+        .catch(handleFlowError(router, "settings", setFlow))
+      return
+    }
+
+    // Otherwise we initialize it
+    ory
+      .createBrowserSettingsFlow({
+        returnTo: String(returnTo || ""),
+      })
+      .then(({ data }) => {
+        setFlow(data)
+        const csrfTokenFromHeaders = data.ui.nodes.find(
+          (node: any) => node.attributes.name === "csrf_token",
+        )?.attributes.value
+        const traits = data.identity.traits
+        setCsrfToken(csrfTokenFromHeaders || "")
+        setTraits(traits)
+        setProfile(traits.additional_information || {})
+      })
+      .catch(handleFlowError(router, "settings", setFlow))
+  }, [flowId, router, router.isReady, returnTo, flow])
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setProfile({
+      ...profile,
+      [event.target.name]: event.target.value,
+    })
+  }
+
+  const onSubmit = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault()
+    const updatedValues = {
+      csrf_token: csrfToken,
+      method: "profile",
+      traits: {
+        ...traits,
+        additional_information: profile,
+      },
+    }
+
+    return (
+      router
+        // On submission, add the flow ID to the URL but do not navigate. This prevents the user losing
+        // his data when she/he reloads the page.
+        .push(`/`)
+        .then(() =>
+          ory
+            .updateSettingsFlow({
+              flow: String(flow?.id),
+              updateSettingsFlowBody: updatedValues,
+            })
+            .then(({ data }) => {
+              // The settings have been saved and the flow was updated. Let's show it to the user!
+              setFlow(data)
+
+              if (data.return_to) {
+                window.location.href = data.return_to
+                return
+              }
+            })
+            .catch(handleFlowError(router, "profile", setFlow))
+            .catch(async (err: AxiosError) => {
+              // If the previous handler did not catch the error it's most likely a form validation error
+              if (err.response?.status === 400) {
+                // Yup, it is!
+                setFlow(err.response?.data)
+                return
+              }
+
+              return Promise.reject(err)
+            }),
+        )
+    )
+  }
+
+  return (
+    <>
+      <Head>
+        <title>Profile Page</title>
```

Comment:
I think we need something generic here that can parse as many Identifiable questionnaires as necessary